servlet-engine/src/main/java/io/micronaut/servlet/engine/DefaultServletHttpRequest.java

```diff
@@ -189,7 +189,7 @@ protected Object buildBody() {
     private Argument<?> resolveBodyType() {
         RouteMatch<?> route = this.getAttribute(HttpAttributes.ROUTE_MATCH, RouteMatch.class).orElse(null);
         if (route != null) {
-            Argument<?> bodyType = route.getBodyArgument()
+            Argument<?> bodyType = route.getRouteInfo().getFullBodyArgument()
```

Comment:
@graemerocher Is this the correct replacement?

Reply:
Seems not (or something else has changed) as now this is failing

https://github.com/micronaut-projects/micronaut-servlet/blob/f23fc5dd2b6cdd3ca25242c366176bf3e037d614/http-server-jetty/src/test/groovy/io/micronaut/servlet/jetty/JettyExceptionHandlerSpec.groovy#L44-L52

Reply:
Looks correct

Reply:
@dstepanov There's also `route.getRouteInfo().getBodyArgument()`... What's the difference?